apps/e2e/tests/js/auth.test.ts

```diff
@@ -0,0 +1,190 @@
+import { it } from "../helpers";
+import { scaffoldProject } from "./js-helpers";
+import { StackClientApp } from '@stackframe/js';
+import { STACK_BACKEND_BASE_URL } from '../helpers';
+import { generateSecureRandomString } from "@stackframe/stack-shared/dist/utils/crypto";
+
+it("should sign up with password", async ({ expect }) => {
+  const { project, app } = await scaffoldProject();
+
+  const clientApp = new StackClientApp({
+    projectId: project.id,
+    baseUrl: STACK_BACKEND_BASE_URL,
+    publishableClientKey: project.publishableClientKey,
+    tokenStore: "memory",
+    urls: {
+      emailVerification: "https://stack-js-test.example.com/verify"
+    }
+  });
+
+  const email = `${crypto.randomUUID()}@stack-js-test.example.com`;
+  const password = generateSecureRandomString();
+
+  const result = await clientApp.signUpWithCredential({
+    email,
+    password,
+  });
+
+  expect(result.status).toBe('ok');
+
+  const user = await clientApp.getUser({
+    or: 'throw',
+  });
+
+  expect(user).toBeDefined();
+  expect(user.primaryEmail).toBe(email);
+});
+
+it("should sign in with password", async ({ expect }) => {
+  const { project, app } = await scaffoldProject();
+
+  const clientApp = new StackClientApp({
+    projectId: project.id,
+    baseUrl: STACK_BACKEND_BASE_URL,
+    publishableClientKey: project.publishableClientKey,
+    tokenStore: "memory",
+    urls: {
+      emailVerification: "https://stack-js-test.example.com/verify"
+    }
+  });
```

Comment:
there are a lot of occurences of you first calling `scaffoldProject`, and then creating a `clientApp`, `serverApp`, or `adminApp`. instead make it so all three of those apps are created in the `scaffoldProject`, and then returned from that function's return object